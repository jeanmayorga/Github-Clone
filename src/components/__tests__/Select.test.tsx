import { render } from '@testing-library/react'
import { Select } from '..'

describe('Render select', () => {
  it('should render select', () => {
    const { container } = render(
      <Select
        name="testSelect"
        options={[
          {
            label: 'Option 1',
            value: 'option-1'
          },
          {
            label: 'Option 2',
            value: 'option-2'
          }
        ]}
        defaultValue="option-1">
        My Select
      </Select>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
