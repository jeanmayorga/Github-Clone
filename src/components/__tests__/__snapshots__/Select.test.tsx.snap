// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render select should render select 1`] = `
<div
  class="mr-1 relative"
>
  <button
    class="border-[#1b1f2426] hover:border-[#1b1f2426] dark:border-[#cdd9e51a] dark:hover:border-[#768390] text-[#24292f] dark:text-[#adbac7] bg-[#f6f8fa] hover:bg-[#f3f4f6] active:bg-[#ebecf0] dark:bg-[#373e47] dark:hover:bg-[#444c56] dark:active:bg-[#3d444d] w-full relative text-sm py-[5px] px-[16px] text-center font-medium whitespace-nowrap cursor-pointer align-middle select-none rounded-md border transition-all disabled:bg-[#f6f8fa] disabled:cursor-default disabled:border-[#1b1f2426]"
  >
    My Select
    <span
      class="inline-block align-middle content-[\\"\\"] border-4 border-b-0 border-r-transparent border-b-transparent border-l-transparent ml-2 opacity-80"
    />
  </button>
  <div
    class="hidden absolute right-0 left-auto top-auto bottom-auto z-30"
  >
    <div
      class="my-3 rounded-md w-[300px] bg-[#ffffff] dark:bg-[#2d333b] border-[#d0d7de] dark:border-[#545d68] border text-xs"
    >
      <header
        class="p-[7px] pl-[16px] flex items-center justify-between border-b border-[#d8dee4] dark:border-[#373e47] font-semibold"
      >
        Select 
        testSelect
        <span
          class="cursor-pointer"
        >
          <svg
            aria-hidden="false"
            aria-label="Close menu"
            class="icon fill-[#57606a] dark:fill-[#768390]"
            data-view-component="true"
            height="16"
            role="img"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </header>
      <div>
        <div
          class="bg-[#eaeef280] dark:bg-[#636e7b1a] text-xs color-[#24292f] dark:color-[#adbac7] px-4 py-[7px] w-full flex border-b last:border-0 border-[#d8dee4] dark:border-[#373e47] hover:bg-[#eaeef280] dark:hover:bg-[#636e7b1a] hover:cursor-pointer transition-all"
        >
          <svg
            aria-hidden="true"
            class="icon mr-2 transition-all dark:fill-[#adbac7]"
            data-view-component="true"
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
              fill-rule="evenodd"
            />
          </svg>
          Option 1
        </div>
        <div
          class="text-xs color-[#24292f] dark:color-[#adbac7] px-4 py-[7px] w-full flex border-b last:border-0 border-[#d8dee4] dark:border-[#373e47] hover:bg-[#eaeef280] dark:hover:bg-[#636e7b1a] hover:cursor-pointer transition-all"
        >
          <svg
            aria-hidden="true"
            class=" invisible icon mr-2 transition-all dark:fill-[#adbac7]"
            data-view-component="true"
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M13.78 4.22a.75.75 0 010 1.06l-7.25 7.25a.75.75 0 01-1.06 0L2.22 9.28a.75.75 0 011.06-1.06L6 10.94l6.72-6.72a.75.75 0 011.06 0z"
              fill-rule="evenodd"
            />
          </svg>
          Option 2
        </div>
      </div>
    </div>
  </div>
</div>
`;
